invokeai/frontend/web/src/features/parameters/hooks/useRecallParameters.ts

```diff
@@ -390,6 +410,113 @@ export const useRecallParameters = () => {
     [prepareLoRAMetadataItem, dispatch, parameterSetToast, parameterNotSetToast]
   );
 
+  /**
+   * Recall ControlNet with toast
+   */
+
+  const { controlnets } = useGetControlNetModelsQuery(undefined, {
+    selectFromResult: (result) => ({
+      controlnets: result.data
+        ? controlNetModelsAdapter.getSelectors().selectAll(result.data)
+        : [],
+    }),
+  });
+
+  const prepareControlNetMetadataItem = useCallback(
+    (controlnetMetadataItem: ControlNetMetadataItem) => {
+      if (!isValidControlNetModel(controlnetMetadataItem.control_model)) {
+        return { controlnet: null, error: 'Invalid ControlNet model' };
+      }
+
+      const {
+        image,
+        control_model,
+        control_weight,
+        begin_step_percent,
+        end_step_percent,
+        control_mode,
+        resize_mode,
+      } = controlnetMetadataItem;
+
+      const matchingControlNetModel = controlnets.find(
+        (c) =>
+          c.base_model === control_model.base_model &&
+          c.model_name === control_model.model_name
+      );
+
+      if (!matchingControlNetModel) {
+        return { controlnet: null, error: 'ControlNet model is not installed' };
+      }
+
+      const isCompatibleBaseModel =
+        matchingControlNetModel?.base_model === model?.base_model;
+
+      if (!isCompatibleBaseModel) {
+        return {
+          controlnet: null,
+          error: 'ControlNet incompatible with currently-selected model',
+        };
+      }
+
+      const controlNetId = uuidv4();
+      const processorType: ControlNetProcessorType =
+        CONTROLNET_MODEL_DEFAULT_PROCESSORS[
+          matchingControlNetModel.model_name
+        ] || initialControlNet.processorType;
+      const processorNode = CONTROLNET_PROCESSORS[processorType].default;
```

Comment:
the processors actually use substrings - so `normal_bae`, for example, should match for the `bae` key. If you look in the controlnetSlice, you can see an example. It doesn't _really_ matter because the image is already processed at this point but could cause confusion if the processor doesn't match the model that was used

Reply:
so should it be CONTROLNET_PROCESSORS[processorType].default.id?

Reply:
Its the `processorType` logic that doesn't work. Look in the controlnetSlice when `controlNetAutoConfigToggled` happens